### PR TITLE
Added ember-hook to components

### PR DESCRIPTION
**Pull request by srowhani**
#patch#

### CODE REVIEW

File: addon/templates/components/nav-route.hbs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {{#if isVisible}}
   {{#if route}}
-    {{#frost-link route disabled=true}}
+    {{#frost-link route disabled=true hook=(concat hook '-link')}}
```
Comment (rox163) [inline addon/templates/components/nav-route.hbs]: What happens if there are multiple 'route' menu items defined in a nav category? How would one distinguish them via the hook?

File: tests/dummy/config/environment.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ module.exports = function (environment) {
     podModulePrefix: 'dummy/pods',
     environment: environment,
     baseURL: '/',
-    locationType: 'hash',
+    locationType: 'auto',
```
Comment (rox163) [inline tests/dummy/config/environment.js]: reason fo r this change back to auto?

File: addon/components/nav-modal.js [hunk 3]
```diff
@@ -31,23 +31,23 @@ export default Component.extend({
...
-    }),
+    }
+  ),
+  init () {
+    this._super(...arguments)
+    ;['outsideClick', 'escape'].forEach(e => {
+      this.actions[e] = function () {
+        this.get('frostNavigation').dismiss()
+      }
+    })
```
Comment (juwara0) [inline addon/components/nav-modal.js]: For lines 44-48, you will run into problems trying to test this code inside init()… Instead, move that into a function that you call inside init(). Then you can call/stub that method to test that it is doing what you expect.

File: tests/dummy/config/environment.js (hunk 2 above)
Comment (srowhani) [inline tests/dummy/config/environment.js]: Forgot about this. Was playing with different browser behaviors in terms of states. It seems they handle popstate / hashchange a bit differently. Not really relevant, I'll revert this bit.

**Comment by rox163** [comment]
README needs an update to show the hooks available. https://github.com/ciena-frost/ember-frost-sort#testing-with-ember-hook

**Comment by srowhani** [comment]
test

File: README.md [hunk 4]
```diff
@@ -172,6 +172,22 @@ Router.map(function () {
...
+| modal hook                                   	| `$hook('frost-nav-modal')`                                                      	|
+| category hook                                	| `$hook('frost-nav-category-<index>')`                                           	|
+| section / column hook                        	| `$hook('frost-nav-modal-section-<index>')`                                      	|
+| section actions hook`                        	| `$hook('frost-nav-modal-section-actions')`                                      	|
+| inline action from section context           	| `$hook('frost-nav-modal-section-<sectionIndex>-action-<actionIndex>')`          	|
+| route hook                                   	| `$hook('frost-nav-modal-section-<sectionIndex>-route-<routeIndex>')`            	|
+| frost-link within the route / action context 	| `$hook('frost-nav-modal-section-<sectionIndex>-(route / action)-<index>-link')` 	|
+| action from section actions context          	| `$hook('frost-nav-modal-section-actions-action-<index>')`                       	|
```
Comment (rox163) [inline README.md]: does the section need an index here as well?

File: README.md [hunk 5]
```diff
@@ -172,6 +172,22 @@ Router.map(function () {
  */
 ```
 
+## Testing with [ember-hook](https://www.npmjs.com/package/ember-hook)
+
+The navigation component is accessible using ember-hook with the top level hook names, or you can access the internal components as well -
+
+| Property                                     	| Hook                                                                            	|
+|----------------------------------------------	|---------------------------------------------------------------------------------	|
+| top level hook                               	| `$hook('frost-nav')`                                                            	|
```
Comment (rox163) [inline README.md]: is this the one used to click() and open the nav?

File: README.md (hunk 4 above)
Comment (rox163) [inline README.md]: I dont like this name....'actions-action'. Can we make it a bit shorter?

File: README.md (hunk 5 above)
Comment (srowhani) [inline README.md]: Nope, that would be category. This is just the div that contains them

Comment (rox163) [inline README.md]: I see. So can we use this one to count and verify the categories displayed?

Comment (srowhani) [inline README.md]: Yep!

**Comment by rox163** [comment]
:+1:

**Pull request merged**